lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_localize_helper.rb

```diff
@@ -171,6 +171,22 @@ def self.verify_pr_diff(main, library, main_strings, lib_strings, source_diff)
             end
           end
         end
+
+        # Extract the value of a import key from build.gradle
+        #
+        # @param [String] import_key The key to look for
+        # @return [String] The value of the key, or nil if not found
+        #
+        def self.get_library_version_from_gradle_config(import_key:)
+          gradle_file_path = File.join(ENV['PROJECT_ROOT_FOLDER'] || '.', 'build.gradle')
+
+          return nil unless File.exists?(gradle_file_path)
+
+          File.open(gradle_file_path, 'r') do | f |
+            text = f.read
+            text.match(/#{import_key}\s?=\s?'(.*?)'/m)&.captures&.first
+          end
```

Comment:
Yeah! I was torn about this, but I ended up thinking that reading the whole file is better in this very specific case. It's not supposed to work on random files, in fact it only works on the `build.gradle` file in the project root folder. The size of this file for our most complex project (WPAndroid) is less than 7KB, so size doesn't seem to be an issue. 
Because of this, I preferred the simpler code here (and I'd bet that modern OSes will actually read files of this size in one step anyway at a lower level to optimize for I/O operations as the size is close to a single page on modern devices).